scripts/test/Muon/run_string_utils_test.py

```diff
@@ -0,0 +1,194 @@
+        run_strings = ["1-5", "1,2,3,4,5"]
+        for run_string in run_strings:
+            run_list = utils.run_string_to_list(run_string)
+            self.assertEqual(run_list, [1, 2, 3, 4, 5])
+
+    def test_run_string_to_list_sorts_list_ascending(self):
+        run_strings = ["5,4,3,2,1", "4,5,1-3"]
+        for run_string in run_strings:
+            run_list = utils.run_string_to_list(run_string)
+            self.assertEqual(run_list, [1, 2, 3, 4, 5])
+
+    def test_run_string_to_list_removes_duplicates(self):
+        run_strings = ["5,5,5,5,4,3,2,1,1", "1,2,4,5,1-3,1-5,1-4"]
+        for run_string in run_strings:
+            run_list = utils.run_string_to_list(run_string)
+            self.assertEqual(run_list, [1, 2, 3, 4, 5])
+
+    def test_run_string_to_list_throws_for_incorrectly_placed_range_separator(self):
+        run_strings = ["-1,2,3", "1,2,3-"]
+        for run_string in run_strings:
+            with self.assertRaises(IndexError) as context:
+                utils.run_string_to_list(run_string)
+                self.assertTrue(run_string + " is not a valid run string" in context.exception)
+
+    def test_run_string_to_list_throws_for_incorrectly_placed_delimiter(self):
+        run_strings = [",1,2,3", "1,2,3,"]
+        for run_string in run_strings:
+            with self.assertRaises(IndexError) as context:
+                utils.run_string_to_list(run_string)
+                self.assertTrue(run_string + " is not a valid run string" in context.exception)
+
+    def test_run_string_to_list_handles_non_consecutive_runs(self):
+        run_lists = [[1, 3, 4, 5], [0, 1, 2, 3, 8, 9, 199, 200]]
+        run_strings = ["1,3-5", "0-3,8-9,199-200"]
+        for i in range(len(run_lists)):
+            self.assertEqual(utils.run_string_to_list(run_strings[i]), run_lists[i])
+
+    def test_run_string_to_list_orders_non_consecutive_runs(self):
+        run_string = "1-3,48-50"
+        run_list = [1, 2, 3, 48, 49, 50]
+        self.assertEqual(utils.run_string_to_list(run_string), run_list)
+
+
+class RunStringUtilsIncrementAndDecrementRunLists(unittest.TestCase):
+
+    def test_cannot_increment_or_decrement_by_zero_or_negative_numbers(self):
+        run_list = [1, 2, 3, 4, 5]
+        self.assertEqual(utils.increment_run_list(run_list, 0), run_list)
+        self.assertEqual(utils.increment_run_list(run_list, -1), run_list)
+        self.assertEqual(utils.decrement_run_list(run_list, 0), run_list)
+        self.assertEqual(utils.decrement_run_list(run_list, -1), run_list)
+
+    def test_can_correctly_increment_run_list_by_one(self):
+        run_list = [1, 2, 3, 4, 5]
+        increment = 1
+        incremented_run_list = [1, 2, 3, 4, 5, 6]
+        self.assertEqual(utils.increment_run_list(run_list, increment), incremented_run_list)
+
+    def test_can_correctly_decrement_run_list_by_one(self):
+        run_list = [2, 3, 4, 5]
+        increment = 1
```

Comment:
should this be decrement?

Reply:
Yes... It's a tricky one, because increment can be used as a noun; and so in this case by increment I mean "the amount by which something can be incremented *or* decremented". Elsewhere increment is used as a verb. Decrement cannot be used as a noun in the same way so I didn't want to change it. But it's clearly confusing so I'll change it to decrement.